include/nil/crypto3/hash/detail/poseidon/poseidon_mds_matrix.hpp

```diff
@@ -80,6 +82,15 @@ namespace nil {
 #endif
                     inline mds_matrix_type generate_mds_matrix() {
                         mds_matrix_type new_mds_matrix;
+                        if (policy_type::full_rounds == 55) { // TODO: kimchi specialization
```

Comment:
This is a draft switch between neptun/kimchi versions. We definitely need something else.

Reply:
I have refactored most of the code, please review again.